sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -418,6 +419,13 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
 
   override def toString: String = treeString
 
+  def toOriginString: String =
+    if (this.origin.callSite.isDefined && !this.isInstanceOf[BoundReference]) {
```

Comment:
When I write the following programs, each of comments for ```Expression``` is what I wanted (```"i <=3"``` is referenced in ```sql```. ```"i"``` is called in ```selectExpr```).
I think that you may want to add a comment to some particular ```BoundReference``` (originally was ```AttributeReference```, in addition to ```Expression```? Is this correct?


```java
sqlContext.sql("select i, f from tempTable WHERE i <= 3 and f > 1.0").show()

/* (input[1, int] <= 3) @ sql at Dataframe1.scala:13 */
```

```java
df.selectExpr("i", "f")

/* isnull(input[0, int]) @ selectExpr at Dataframe1.scala:13 */
```